docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,208 @@
+# RFC: Stop pre-processing styles with Griffel in `@fluentui/react-components`
+
+[@layershifter](https://github.com/layershifter)
+
+## Background
+
+We've begun noticing a pattern where apps are bundled into multiple separate bundles. For example, there's the main bundle for the app itself and a separate bundle for a first UI piece fetched from a CDN. This problem is described in the following issues:
+
+- [microsoft/griffel#453](https://github.com/microsoft/griffel/issues/453) (_check this issue for more details_)
+- [microsoft/griffel#526](https://github.com/microsoft/griffel/issues/526)
+
+_TL;DR_: Griffel relies on the order of CSS classes, which can lead to clashes when multiple bundles are involved. See the simple example below:
+
+```html
+<!-- CURRENT STATE -->
+
+<!-- main bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+  .order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  🔴 We expect that a "div" below will have "padding-left: 5px", but instead, it has "padding: 10px".
+
+     The issue appears because the CDN bundle loads after the main bundle, resulting in style overrides. This occurs because ".order0" appears in both bundles, and CSS prioritizes the order of appearance. 
+  -->
+
+<div class="order0 order1"></div>
+```
+
+The single reliable solution is to prefix styles with a unique identifier. This prevents CSS rules from colliding, for example:
+
+```html
+<!-- PROPOSAL -->
+
+<!-- main bundle -->
+<style>
+  .main-order0 {
+    padding: 10px;
+  }
+  .main-order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  ✅ Now the "div" below will have "padding-left: 5px" as CSS rules don't clash anymore.
+  -->
+
+<div class="main-order0 main-order1"></div>
+```
+
+The proposed idea is to integrate prefix as a part of classes generation in Griffel as a salt for hashing (see [`hashClassName`](https://github.com/microsoft/griffel/blob/e3bf466e36464585ecff0ca3492760f6ada76dcc/packages/core/src/runtime/utils/hashClassName.ts#L23-L32) function):
```

Comment:
Just to make sure I'm understanding correctly, the prefix will be an input into the hash now? So:

```ts
const classNameWithHashedPrefix == hash(stuffWeAlreadyUseToHash + prefix);
```

> 👍 Apps can easily _prefix_ classes with a unique identifier like a project name

And it will also be possible to generate a class name like:

```css
.my-prefix-$hashedClassName ...
```

Assuming my understanding above is correct, is it possible to use both together? E.g.:

```css
.my-prefix-$classNameWithHashedPrefix 
```

Reply:
Thanks for the clarification!